app/src/main/java/com/tomaszpolanski/androidsandbox/MainActivity.kt

```diff
@@ -0,0 +1,53 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.os.Bundle
+import android.widget.Toast
+import com.tomaszpolanski.androidsandbox.Event.LongClick
+import com.tomaszpolanski.androidsandbox.Event.ShortClick
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : ActivityIntent<State, Event>() {
+
+    private lateinit var hiddenTip: Toast
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setSupportActionBar(toolbar)
+        hiddenTip = Toast.makeText(this, "", Toast.LENGTH_SHORT)
+
+        viewModel = pretendedInjection()
+        bind(savedInstanceState, State.HiddenToast(0))
```

Comment:
Because the ViewModel survives configuration changes, won't this reapply the state held in the bundle to the existing ViewModel. What happens when the state is significantly more complex - such as a list of entries in a RecyclerView? Perhaps the ViewModel should hold that content and reinitialize itself?

Perhaps for the first iteration it might be worth disregarding the bundle?

More thoughts on this issue here: https://proandroiddev.com/customizing-the-new-viewmodel-cf28b8a7c5fc

Reply:
Handling the bundle is pretty useful concept, I would like to keep it and experiment with it